control-plane-operator/controllers/hostedcontrolplane/manifests/etcd.go

```diff
@@ -105,3 +109,68 @@ func EtcdBackupCronJob(hcpNamespace string) *batchv1.CronJob {
 		},
 	}
 }
+
+// resourceNameForShard generates resource names for etcd shards
+// For backward compatibility, the default shard uses the base name without suffix
+// Named shards use the pattern: baseName-shardName
+func resourceNameForShard(baseName, shardName string) string {
```

Comment:
This is duplicated in at least one other location in this PR, is there a way to avoid the duplication/possibility of drift